panics/panics.go

```diff
@@ -0,0 +1,43 @@
+package panics
+
+import (
+	"fmt"
+	"runtime/debug"
+)
+
+// CallBackFn is a function that will get called with information about the panic
+type CallBackFn func(recoverObj interface{}, debugStackTrace string)
+
+// PanicHandler is a function that can be called with defer to recover from panics and pass them to a callback
+// it returns an error if a recovery was needed
+type PanicHandler func() error
+
+// MakeHandler makes a handler that recovers from panics and passes them to the given callback
+func MakeHandler(cb CallBackFn) PanicHandler {
+	return func() error {
+		obj := recover()
+		if obj == nil {
+			return nil
+		}
+		stack := string(debug.Stack())
+		if cb != nil {
+			cb(obj, stack)
+		}
+		return RecoveredPanicErr{
+			PanicObj:        obj,
```

Comment:
is it worth the separate package / factory / complexity for calling recover in one place?